lib/diagnostic.js

```diff
@@ -5,29 +5,56 @@
 // and returns the minimum from that set
 // assume that all the argument are numbers
 const min = function min() {
-
+  let result = arguments[0];
+  for (let i = 1; i < arguments.length; i++) {
+    if (result > arguments[i]) {
+      result = arguments[i];
+    }
+  }
+  return result;
 };
 
 // Question 2
 // Write a function that takes zero or more arguments
 // and returns the sum of that set
 // assume that all the argument are numbers
 const sum = function sum() {
-
+  let result = 0;
+  for (let i = 1; i < arguments.length; i++) {
+    result = arguments[i] + result;
+  }
+  return result;
 };
 
 // Question 3
 // Write a function that returns an array of the specified length
 // with all the elements set to defaultValue
 const newArray = function newArray(length, defaultValue) {
-
+  let array = [defaultValue];
+  for (let i = 1; i < length; i++) {
+    let array += defaultValue;
```

Comment:
You have already declaired `array`. line 35 is breaking the `grunt test` when we try to run the test script. Be sure to run `grunt test` as the readme states.